### DISCUSSION TITLE
How does one get the generator going?

### DISCUSSION THREAD

**Discussion by AndrewAtAvenza** [cross-language-cpp/djinni-generator#151] [Q&A] [answered]
We've been using the original djinni for a while and I was delighted to find that was a project that had picked up the torch. I've finally found the time to try and migrate to the this version of the project but I've run into a rather firm wall: I have no idea how to build/use this version of the generator.

Our old djinni repo has a `run` script that we can feed all the familiar djinni arguments but I cannot figure out what I'm supposed to do with the djinni-generator release I've downloaded. Admittedly I know absolutely nothing about Scala, so that maybe the problem. I gather I probably need to use `sbt`? Some of the stuff I've read imply its included with the JDK but it doesn't seem to be present in the JRE that's included with Android Studio (via the JetBrains Toolbox). I installed `sbt` but I'm no closer to figuring out what I should do. 

There don't actually seem to be any instructions on setting up the repo and I don't know if that's an oversight or if it's supposed to be obvious what I should do :) I'm sure once I get it to the point where it I can run the generator I'll be fine, but can someone guide in me in whatever steps I need to take to make the 1.3.2 download I've setup functional?

Any help is greatly appreciated!

**Comment by a4z** [comment, marked as answer]
Hi Andrew

You can download the djinni artifact and run it. 
All it needs is a Java installation on your computer
https://github.com/cross-language-cpp/djinni-generator/releases

Here are additional ways of installation listed, for the various operating systems.
https://djinni.xlcpp.dev/djinni-generator/setup/
(You forgot to mention on which one you are)

So you do not need to clone the repo and build it.
But if you want, have a look at the [pipeline file](https://github.com/cross-language-cpp/djinni-generator/blob/main/.github/workflows/main.yaml) to see how it can be built 

Hope that helps you.

**Comment by AndrewAtAvenza** [comment]
I tried downloading that but since the link was 'djinni.bat' and what I got was _clearly_ not a batch file I assumed it was some kind of glitch.

I just tried dropping in to our old build pipeline and _man_, it works fantastic! It clearly runs faster than before and eliminates a lot of nasty warnings to boot. I should have updated ages ago!

Thanks so much! We were using the djinni support library in a weird crusty way so I'll set it up properly this time with `djinni-support-lib` and everything should be _much_ nicer.

Thanks again!

**Reply by a4z** [reply]
Yes, on Windows, it needs to be a bat file to be executable and do its magic. 
Maybe one day, I will find time to add a [scoop package](https://scoop.sh), which would greatly simplify the Windows installation.

Happy you find it useful what we do. Thanks for the feedback!
